src/library/dlgtagfetcher.cpp

```diff
@@ -250,45 +299,63 @@ void DlgTagFetcher::fetchTagFinished(
 }
 
 void DlgTagFetcher::slotNetworkResult(
-        int httpError, const QString& app, const QString& message, int code) {
+        int httpError,
+        const QString& app,
+        const QString& message,
+        int code) {
     m_networkResult = httpError == 0 ? NetworkResult::UnknownError : NetworkResult::HttpError;
     m_data.m_pending = false;
-    QString strError = tr("HTTP Status: %1");
-    QString strCode = tr("Code: %1");
-    httpStatus->setText(strError.arg(httpError) + "\n" + strCode.arg(code) + "\n" + message);
-    QString unknownError = tr("Mixxx can't connect to %1 for an unknown reason.");
-    cantConnectMessage->setText(unknownError.arg(app));
-    QString cantConnect = tr("Mixxx can't connect to %1.");
-    cantConnectHttp->setText(cantConnect.arg(app));
+
+    if (m_networkResult == NetworkResult::UnknownError) {
+        QString unknownError = tr("Can't connect to %1 for an unknown reason.");
+        loadingProgressBar->setFormat(unknownError.arg(app));
+    } else {
+        QString cantConnect = tr("Can't connect to %1.");
+        loadingProgressBar->setFormat(cantConnect.arg(app));
+    }
+
+    btnRetry->setEnabled(true);
     updateStack();
 }
 
 void DlgTagFetcher::updateStack() {
+    m_progressBarValue = 10;
+    loadingProgressBar->setValue(m_progressBarValue);
+
+    btnApply->setDisabled(true);
+
+    successMessage->setVisible(false);
+    loadingProgressBar->setVisible(true);
+
+    results->clear();
+    addDivider(tr("Original tags"), results);
+    addTrack(trackColumnValues(*m_track), -1, results);
```

Comment:
What does -1 mean here?

Reply:
-1 is the index of the actual track metadata. The results fetched from musicbrainz starts from index 0.

Reply:
Ok please use a constant for this. Something like
kOriginalTrackIndex or such.